packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -159,12 +164,14 @@ export class S3Provider implements StorageProviderInterface {
       ? config.aws.s3.endpoint
         ? `${config.aws.s3.endpoint.replace('http://', '').replace('https://', '')}/${this.bucket}`
         : config.aws.cloudfront.domain
+      : config.server.storageProvider === 's3-do'
+      ? config.aws.cloudfront.domain
       : `${config.aws.cloudfront.domain}/${this.bucket}`
 
   originURLs = [this.cacheDomain]
 
-  private bucketAssetURL =
-    config.server.storageProvider === 's3'
+  public bucketAssetURL =
```

Comment:
All the `'s3-do'` specific check in this file should be moved to S3DOStorageProvider. The class variable should be updated based on DO in constructor of it.